Add ListCardComponent and MapCardComponent with Associated Styles and Data for the BEAT plan.

KarthikShetty818 opened the pull request:
###  Add ListCardComponent and MapCardComponent with Associated Styles and Data for the BEAT plan. 

- Added 'UserImg.png' to load on each card if the user does not have a profile image.
- Added ListCardComponent to render items using FlatList in MyBeatPlan.screen.tsx.
- Included MapCardComponent to render horizontally scrollable items using FlatList in MyBeatPlan.screen.tsx.
- Updated dummyData.ts to include sample data for beatListCardData.
- Imported NavIconImg.svg, phoneGreenImg.svg, UserDefaultImg.svg, and beatProfileImg.png assets.
- Created ListCardComponent and associated style file listCardComponent.style.ts.
- Implemented ListCardComponent to display card details including name, address, distance, time, and image.
- Created MapCardComponent and associated style file mapCardComponent.style.ts.
- Implemented MapCardComponent to display card details including count, name, address, distance, time, and navigation link.
- Ensured all components are responsive and styled consistently with theme colors.

KarthikShetty818 committed:
Add ListCard and MapCard Components with Associated Styles and Data for the BEAT plan.

KarthikShetty818 committed:
Minor Fixes

kartikDeloitte commented on app/screens/beat/components/listCardComponent/listCardComponent.style.ts:
```diff
@@ -0,0 +1,87 @@
+  container: {
+    padding: 10,
+    marginVertical: 5,
+    backgroundColor: '#ffffff',
```
use from COLORS

kartikDeloitte commented on app/screens/beat/components/listCardComponent/listCardComponent.style.ts:
```diff
@@ -0,0 +1,87 @@
+    padding: 10,
+    marginHorizontal: 5,
+  },
+  nameText: {
```
use variant and text from react native paper

kartikDeloitte commented on app/screens/beat/components/listCardComponent/listCardComponent.style.ts:
```diff
@@ -0,0 +1,87 @@
+  },
+  addressText: {
+    fontSize: 12,
+    color: '#796767',
```
use from COLORS

kartikDeloitte commented on app/screens/beat/components/listCardComponent/listCardComponent.style.ts:
```diff
@@ -0,0 +1,87 @@
+    marginLeft: 18,
+    padding: 5,
+    fontSize: 12,
+    color: '#656767',
```
COLORS

kartikDeloitte commented on app/screens/beat/components/listCardComponent/listCardComponent.tsx:
```diff
@@ -0,0 +1,58 @@
+
+function ListCardComponent({ image, name, address, distance, time }: ListCardComponentProps) {
+  return (
+        <View style={styles.container}>
```
use card component already present in components

kartikDeloitte commented on app/screens/beat/components/mapCardComponent/mapCardComponent.style.ts:
```diff
@@ -0,0 +1,106 @@
+    paddingTop: 5,
+    marginVertical: 10,
+    marginHorizontal: 5,
+    backgroundColor: '#ffffff',
```
remove COLORS from every where

kartikDeloitte commented on app/screens/beat/components/mapCardComponent/mapCardComponent.tsx:
```diff
@@ -0,0 +1,68 @@
+
+function MapCardComponent({ count, name, address, distance, time }: MapCardComponentProps) {
+  return (
+        <View style={styles.container}>
```
use card component from already made components

kartikDeloitte commented on app/utils/dummyData.ts:
```diff
@@ -404,3 +404,211 @@ export const orderDashboard = {
     ],
   },
 };
+
```
why have you added this code?

kartikDeloitte commented:
create a component folder in beatPlan and the component there.

KarthikShetty818 committed:
Minor Fixes

KarthikShetty818 commented:
### [1deb926] Implemented the changes 

- Update color codes to use values from COLORS file
- Wrap card component with card wrapper
- Apply text variants

KarthikShetty818 committed:
Create CommonCard Component for ListCard and MapCard Components.

KarthikShetty818 commented:
### [5a0ae69] Create CommonCard Component for ListCard and MapCard Components.

•	Created CommonCard component to encapsulate shared structure and styles between ListCard and MapCard components.
•	Ensured adherence to DRY (Don't Repeat Yourself) principles by moving common code to a single reusable component.
•	Updated ListCard and MapCard components to utilize the new CommonCard component.
•	Added UserImg.png as the default profile image for users without a profile picture.
•	Implemented ListCard component to render items using FlatList in MyBeatPlan.screen.tsx.
•	Included MapCard component to render horizontally scrollable items using FlatList in MyBeatPlan.screen.tsx.
•	Updated dummyData.ts to include sample data for beatListCardData.
•	Imported necessary assets: NavIconImg.svg, phoneGreenImg.svg, UserDefaultImg.svg, and beatProfileImg.png.
•	Ensured all components are responsive and styled consistently with theme COLORS.
•	Deleted 2 folders ‘listCardComponent’ and ‘mapCardComponent’ in the ‘app/screens/beat/components’ as they were not required.
